DB/common.php

```diff
@@ -1873,7 +1873,7 @@ function dropSequence($seq_name)
      *
      * @see PEAR_Error
      */
-    function &raiseError($code = DB_ERROR, $mode = null, $options = null,
+    static function raiseError($code = DB_ERROR, $mode = null, $options = null,
```

Comment:
Poking at this further... 
 * It looks like `DB_common::raiseError($code, $mode,...)` has a rather different signature from the parent's `PEAR::raiseError($message,$code,$mode,...)`. E-v-i-l. Ee-vv-ii-ll. Eee-vvv-iii-lll.
 * PHP allows you to call `static` function using non-static notation (e.g. `$object->myStaticFunc()`), although `$this` does not propagate.
 * By adding `static` and obscuring `$this`, the patch actually makes `DB_common::raiseError()` behave differently -- it no longer reports the failed SQL query!
 * I can imagine why it was necessary to add `static` here when using PEAR 1.9 (i.e. there was a conflict in the function signatures); but in PEAR 1.10, upstream `PEAR::raiseError()` is actually a magic function which means it still runs even if the class-hierarchy suffers from conflicted signatures.